tests/test_helper.hpp

```diff
@@ -232,3 +247,33 @@ void check_results(std::array<T, NumElements> output,
     check_results(output[i], reference[i], tol_multiplier);
   }
 }
+
+template <typename T>
+struct SyclFpMatcher : Catch::Matchers::MatcherGenericBase {
+  using FpType = decltype((T{}).real());
+  SyclFpMatcher(T target, int tol_multiplier = 1)
+      : target_(target), tol_multiplier_(tol_multiplier) {}
+
+  template <typename OtherT> bool match(OtherT const &other) const {
+    return detail::almost_equal(other, target_,
```

Comment:
Can't you use ```static_cast``` to std::complex< ```value_type``` of T>? Making sure that OtherT has the same.

That's what I did here: https://github.com/argonne-lcf/SyclCPLX/blob/80e1c54cd26f653ec495ac4852c6a93784b8837c/tests/test_helper.hpp#L205

Removing the need to have the ```almost_equal```'s specialization.

Tests have been added to check that the result of the cast is still equal to the original sycl::complex and let us safely simplify test_helper.

Reply:
I modified ComplexMatcher implementation to convert to std::complex, avoiding the need to add back a sycl cplx specialization, which I think is cleaner.